Fabric.Authorization.API/Modules/UsersModule.cs

```diff
@@ -57,26 +57,52 @@ public class UsersModule : FabricModule<User>
             return groups;
         }
 
-        private async Task<dynamic> AddGranularPermissions(dynamic param, bool denied)
+        private async Task<dynamic> AddGranularPermissions(dynamic param)
         {
-            var apiModel = this.Bind<GranularPermissionApiModel>();
-
-            if (apiModel.Target != param["subjectId"])
-                return CreateFailureResponse("Target must be the subject id.", HttpStatusCode.BadRequest);
+            var apiModel = this.Bind<GranularPermissionApiModel>();            
 
             foreach (var perm in apiModel.Permissions)
                 await CheckAccess(_clientService, perm.Grain, perm.SecurableItem, AuthorizationManageClientsClaim);
 
             var granularPermissions = apiModel.ToGranularPermissionDomainModel();
 
-            if (denied)
-                granularPermissions.DeniedPermissions = apiModel.Permissions.Select(p => p.ToPermissionDomainModel());
-            else
-                granularPermissions.AdditionalPermissions =
-                    apiModel.Permissions.Select(p => p.ToPermissionDomainModel());
+            granularPermissions.IdentityProvider = param.identityProvider;
+            granularPermissions.Id = param.subjectId;
+
+            granularPermissions.DeniedPermissions = apiModel.Permissions

```

Comment:
If I'm not mistaken, clients will have to pass in 0 for "add" and 1 for "deny" since this is an enum. Should we make it a string so that it's easier for clients to remember? I noticed this with an enum I was using in a response and it was sending back integers for the enum.

Reply:
Interesting. If thats the case then I will change it so the clients can pass in a string